File: KITTY/KURS_1/JAVA/DAY_20_KEYBOARD/app/src/main/java/com/example/velic/day_20_keyboard/BlankFragment.java
package com.example.velic.day_20_keyboard;

import android.content.Context;
import android.net.Uri;
import android.os.Bundle;

import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.Fragment;

import android.text.InputType;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.EditText;


/**
 Если продолжить тему показа клавиатуры, то может возникнуть следующая ситуация.
 Допустим у вас есть DialogFragment с EditText.
 При выводе диалогового окна вам нужно установить фокус на EditText
 и показать клавиатуру:
 */



public class BlankFragment extends DialogFragment {

    private EditText editText;

    public BlankFragment() {
    //     Empty constructor required for DialogFragment
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_blank, container);
        editText = (EditText) view.findViewById(R.id.   txt_yourName);

        // Request focus and show soft keyboard automatically
        editText.requestFocus();
        getDialog().getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE);
       // Когда элемент EditText получает фокус, то появляется клавиатура. Можно установить нужный вид клавиатуры через атрибут InputType или программно через метод setInputType():

        EditText ipt = editText; //new EditText(this);
        ipt.setInputType(InputType.TYPE_CLASS_PHONE);  //установит клавиатуру для ввода номера телеф

        return view;
    }
}

File: KITTY/KURS_1/KOTLIN/DAY_19_1_Keyboard_kt/app/src/main/java/com/example/day_19_1_keyboard_k/BlankFragment.kt
package com.example.day_19_1_keyboard_k

import android.os.Bundle
import android.text.InputType
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import android.widget.EditText

import androidx.fragment.app.DialogFragment
import kotlinx.android.synthetic.main.fragment_blank.*


/**
 * Если продолжить тему показа клавиатуры, то может возникнуть следующая ситуация.
 * Допустим у вас есть DialogFragment с EditText.
 * При выводе диалогового окна вам нужно установить фокус на EditText
 * и показать клавиатуру:
 */


class BlankFragment : DialogFragment() {

    // private var editText: EditText? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        //val view = inflater.inflate(R.layout.fragment_blank, container)
       // editText = view.findViewById<View>(R.id.txt_yourName) as EditText

        // Request focus and show soft keyboard automatically
        txt_yourName.requestFocus()
        dialog?.window!!.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE)
        // Когда элемент EditText получает фокус, то появляется клавиатура. Можно установить нужный вид клавиатуры через атрибут InputType или программно через метод setInputType():

        val ipt = txt_yourName //new EditText(this);
        ipt!!.inputType = InputType.TYPE_CLASS_PHONE  //установит клавиатуру для ввода номера телеф

        return view
    }
}//     Empty constructor required for DialogFragment